src/config_waiter.rs

```diff
@@ -64,19 +68,34 @@ pub fn wait_for_partition_boot(config_store: Arc<RwConfigStore>, partition: u16)
             .all_configs()
             .len();
 
-        tracing::info!(
-            boot_time_ms,
-            total_configs,
-            partition,
-            "config_consumer.partition_boot_complete",
-        );
-
-        metrics::gauge!("config_consumer.partition_boot_time_ms", "partition" => partition.to_string())
-            .set(boot_time_ms as f64);
-        metrics::gauge!("config_consumer.partition_total_configs", "partition" => partition.to_string())
-            .set(total_configs as f64);
+        let boot_string = format!(
+            "config_consumer.partition_boot_{}",
+            if shutdown.is_cancelled() {
+                "cancelled"
+            } else {
+                "complete"
+            }
+        )
+        .to_string();
+        tracing::info!(boot_time_ms, total_configs, partition, boot_string);
+
+        if !shutdown.is_cancelled() {
+            tracing::info!(
+                boot_time_ms,
+                total_configs,
+                partition,
+                "config_consumer.partition_boot_complete",
+            );
+
+            metrics::gauge!("config_consumer.partition_boot_time_ms", "partition" => partition.to_string())
+                .set(boot_time_ms as f64);
+            metrics::gauge!("config_consumer.partition_total_configs", "partition" => partition.to_string())
+                .set(total_configs as f64);
+        }
 
-        boot_finished.send(()).expect("Failed to report boot");
+        boot_finished
+            .send(!shutdown.is_cancelled())
+            .expect("Failed to report boot");
```

Comment:
So my thought here was to just early return, so right after the while loop you could do

```
if shutdown.is_cancelled() {
	boot_finished
			.send(false)
			.expect("Failed to report boot");
	return;
}
```

(or alternatively if we have the enum)

```rs
if shutdown.is_cancelled() {
	boot_finished
			.send(BootResult::Canceled)
			.expect("Failed to report boot");
	return;
}
```


Then we don't have to put any of the tracing stuff in an if block